corehq/apps/userreports/reports/filters/values.py

```diff
@@ -128,35 +133,129 @@ def to_sql_values(self):
         }
 
 
+def get_dyn_range_filter(date_range_slug):
+
+    class DynRangeFilter(BasicFilter):
+        date_range = date_range_slug
+
+        def build_expression(self, table):
+            from sqlalchemy import bindparam
+            start_date, end_date = get_daterange_start_end_dates(self.date_range, *self.parameter)
+            return get_column(table, self.column_name).between(
+                bindparam(start_date), bindparam(end_date)
+            )
+
+    return DynRangeFilter
+
+
 class PreFilterValue(FilterValue):
 
+    class BasicBetweenFilter(BasicFilter):
+        # We want a BetweenFilter with the same __init__ params as INFilter
+        def build_expression(self, table):
+            assert len(self.parameter) == 2
+            from sqlalchemy import bindparam
+            return get_column(table, self.column_name).between(
+                bindparam(self.parameter[0]), bindparam(self.parameter[1])
+            )
+
+    dyn_operator_filters = {
+        c.slug: get_dyn_range_filter(c.slug) for c in get_all_daterange_choices()
+    }
+    null_operator_filters = {
+        '=': ISNULLFilter,
+        '!=': NOTNULLFilter,
+        'is': ISNULLFilter,
+        'is not': NOTNULLFilter,
+    }
+    array_operator_filters = {
+        'in': INFilter,
+        'between': BasicBetweenFilter,
+    }
+    scalar_operator_filters = {
```

Comment:
this looks like an exact copy of `NumericFilterValue.operators_to_filters`. i'd suggest combining the two.